shm/src/Store.cpp

```diff
@@ -45,13 +48,76 @@ Store::Response Store::sell(Cargo* cargo, size_t amount, Player* player) {
 }
 
 void Store::nextDay() {
-    std::mt19937 generator(std::random_device{}());
-    std::uniform_int_distribution<size_t> distribution{
-        MIN_CARGO_IN_STORE, MAX_CARGO_IN_STORE
-    };
-
     for (const auto& cargo : cargo_) {
-        cargo->setAmount(distribution(generator));
+        cargo->setAmount(randomGenerate(MIN_CARGO_IN_STORE, MAX_CARGO_IN_STORE));
     }
 }
 
+size_t Store::randomGenerate(size_t min, size_t max) {
+    std::mt19937 gen(std::random_device{}()); 
+    std::uniform_int_distribution<size_t> distribution(min, max);
+    return distribution(gen);
+}
+
+Item::Rarity Store::rarityCoversion(size_t numberForConversion) {
+    size_t multiplyRarity = 4;
+    size_t addRarity = 1;
+    return static_cast<Item::Rarity>((numberForConversion * multiplyRarity) + addRarity);
+}
+
+void Store::createAllCargo() {
+    cargo_.reserve(40);
+    createFruits();
+    createAlcohols();
+    createItems();
+    createDryFruits();
+}
+
+void Store::createFruits() {
+    cargo_.push_back(std::make_unique<Fruit>(Fruit("Orange", randomGenerate(0, 100), randomGenerate(10, 20))));
```

Comment:
It's quite expensive to call 'randomGenerate' which initializes random device, random generator and uniform distribution every time it is called only to return a single number. Maybe  a better idea would be to hardcode these values into a file and read them in one simple loop to fill the vector.